src/utils/XhrNetworkInterceptor.ts

```diff
@@ -98,26 +99,33 @@ export const injectHeaders = async (
 
   const headerFound = networkData.requestHeaders.traceparent != null;
 
-  networkData.w3cc = headerFound;
-  headerFound
+  networkData.isW3cHeaderFound = headerFound;
+  const injectionMethodology = headerFound
     ? IdentifyCaughtHeader(networkData, w3c_caught_header)
     : injectGeneratedData(networkData, w3c_generated_header);
+  return injectionMethodology;
 };
 
-const IdentifyCaughtHeader = (networkData: NetworkData, w3c_caught_header: boolean) => {
-  w3c_caught_header && (networkData.wceti = networkData.requestHeaders.traceparent);
+const IdentifyCaughtHeader = async (networkData: NetworkData, w3c_caught_header: boolean) => {
+  if (w3c_caught_header) {
+    networkData.W3CCaughtHeader = networkData.requestHeaders.traceparent;
+    return networkData.requestHeaders.traceparent;
+  } else {
+    return;
+  }
 };
 
-const injectGeneratedData = (networkData: NetworkData, w3c_generated_header: boolean) => {
+const injectGeneratedData = async (networkData: NetworkData, w3c_generated_header: boolean) => {
   const { timestampInSeconds, partialId, w3cHeader } = generateW3CHeader(networkData.startTime);
+  networkData.partialId = partialId;
+  networkData.networkStartTimeInSeconds = timestampInSeconds;
 
   if (w3c_generated_header) {
-    networkData.wgeti = w3cHeader;
-    networkData.requestHeaders.traceparent = w3cHeader;
+    networkData.W3CgeneratedHeader = w3cHeader;
+    return w3cHeader;
+  } else {
+    return;
   }
```

Comment:
Same for the `else` block here as well, we can just remove it as there's no logic after it.